dotCMS/src/main/java/com/dotmarketing/business/cache/provider/h22/H22Cache.java

```diff
@@ -509,89 +544,51 @@ private Object doSelect(Fqn fqn) throws Exception {
 			return null;
 		}
 
-		ObjectInputStream input=null;
-		InputStream bin=null;
-		InputStream is=null;
+
 		Optional<Connection> opt = createConnection(true, db(fqn));
 		if (!opt.isPresent()) {
 			return null;
 		}
-		Connection c = opt.get();
-
-		PreparedStatement stmt = null;
-		try {
-
-			stmt = c.prepareStatement("select CACHE_DATA from `" + TABLE_PREFIX + table(fqn) + "` WHERE cache_id = ?");
+        try(Connection c = opt.get();
+            PreparedStatement stmt = c.prepareStatement("select CACHE_DATA from `" + TABLE_PREFIX + table(fqn) + "` WHERE cache_id = ?");){
 			stmt.setString(1, fqn.id);
-			ResultSet rs = stmt.executeQuery();
-			if (!rs.next()) {
-				return null;
+			try(ResultSet rs = stmt.executeQuery()){
+    			if (!rs.next()) {
+    				return null;
+    			}
+    			
+    			try  (final InputStream bin=new BufferedInputStream(new ByteArrayInputStream(rs.getBytes(1)), 8192);
+                       final ObjectInputStream input=new ObjectInputStream(bin);){
+    
+    			    return input.readObject();
+    			}
 			}
-			is = new ByteArrayInputStream(rs.getBytes(1));
-			bin = new BufferedInputStream(is, 8192);
-
-			input = new ObjectInputStream(bin);
-			return input.readObject();
 
-		} finally {
 
-			if (stmt != null) stmt.close();
-			c.close();
-			if (input != null){
-				try {
-					input.close();
-				} catch (IOException e) {
-					Logger.warn(getClass(), "should not be here:" + e.getMessage(),e);
-				}
-			}
-			if (bin != null){
-				try {
-					bin.close();
-				} catch (IOException e) {
-					Logger.warn(getClass(), "should not be here:" + e.getMessage(),e);
-				}
-			}
-			
-			if (is != null){
-				try {
-					is.close();
-				} catch (IOException e) {
-					Logger.warn(getClass(), "should not be here:" + e.getMessage(),e);
-				}
-			}
 		}
 	}
 
 	private void doDelete(Fqn fqn) throws SQLException {
 		if (fqn == null) {
 			return;
 		}
-
+		String sql = "DELETE from " + TABLE_PREFIX + table(fqn) + " WHERE cache_id = ?";
```

Comment:
Codacy found an issue: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5918493)